app/src/main/java/com/worldsnas/starplayer/ExoPlayerService.kt

```diff
@@ -0,0 +1,425 @@
+package com.worldsnas.starplayer
+
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.IBinder
+import android.text.TextUtils
+import androidx.core.app.NotificationCompat
+import com.google.android.exoplayer2.*
+import com.google.android.exoplayer2.source.*
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray
+import com.google.android.exoplayer2.upstream.DefaultHttpDataSourceFactory
+import com.worldsnas.starplayer.model.Music
+import java.util.*
+
+class ExoPlayerService : Service(), Player.EventListener {
+
+
+    lateinit var mediaSource: ConcatenatingMediaSource
+
+    lateinit var player: SimpleExoPlayer
+
+    private var currentWindow = 0
+    private var playbackPosition: Long = 0
+    private var musicList: ArrayList<Music?>? = null
+    private var currentMusicId: Int = 0
+
+
+    private fun reInit() {
+        playbackPosition = 0
+        prepareCurrentWindow(musicList, currentMusicId)
+        initializePlayer()
+    }
+
+
+    override fun onDestroy() {
+        super.onDestroy()
+        releasePlayer()
+    }
+
+
+    private fun play(trackModel: Music?) {
+        val trackModels = ArrayList<Music?>()
+        trackModels.add(trackModel)
+        play(trackModels)
+    }
+
+    private fun play(trackModels: ArrayList<Music?>?) {
+        reInit()
+        if (trackModels != null) {
+            val mediaSources =
+                arrayOfNulls<MediaSource>(trackModels.size)
+            for (i in trackModels.indices) {
+                mediaSources[i] =
+                    buildMediaSource(Uri.parse(trackModels[i]?.address))
+            }
+            mediaSource = ConcatenatingMediaSource(*mediaSources)
+            player.prepare(mediaSource)
+            startPlayer()
+        }
+    }
+
+    private fun prepareCurrentWindow(trackModels: ArrayList<Music?>?, currentMusicId: Int) {
+
+        if (trackModels != null && trackModels.size > 0)
+            for (i in trackModels.indices) {
+                if (trackModels[i]?.id == currentMusicId)
+                    currentWindow = i
+            }
+    }
+
+
+    private fun pausePlayer() {
+        playbackPosition = player.currentPosition
+        currentWindow = player.currentWindowIndex
+        player.playWhenReady = false
+        showCurrentTrackNotification(musicList?.get(currentWindow))
+    }
+
+    private fun startPlayer() {
+        player.seekTo(currentWindow, playbackPosition)
```

Comment:
chandin bar dare `seekTo` seda zade mishe
benazaram yek bar kafi bashe
aslan mage ba play niaz hast?

Reply:
in method 2 bar seda zade mishe :
1-user bare aval ro ye music click mikone(ke baraye tashkhise inke user kodum music ro click karde ke pakhsh beshe niyaz )
 2-user music ro play mikone:(baraye tashkhise pakhshe music as jayee ke pause shode niyaz)

Reply:
method startPlayer() vase action haye(next,prev,play) estefade shode ke baraye har kodum az in action ha niyaz hash player.seekTo seda zade beshe